datasource.go

```diff
@@ -138,36 +138,36 @@ func (ds *SQLDatasource) QueryData(ctx context.Context, req *backend.QueryDataRe
 
 }
 
-func (ds *SQLDatasource) getDBConnectionFromQuery(q *Query, datasourceUID string) (string, dbConnection, error) {
+func (ds *SQLDatasource) GetDBConnectionFromQuery(q *Query, datasourceUID string) (string, *sql.DB, *backend.DataSourceInstanceSettings, error) {
```

Comment:
mmh, from the usage I see [here](https://github.com/grafana/grafana-aws-sdk/pull/61/files#diff-b49eb8778b44eee530db57eea65f712ef48d4194da9e7fa96016a25d28f4728dR152), you are only interested on getting the `*sql.DB`, not the internal key and the instance settings. I would create two different functions here: `GetDBFromQuery` that returns what you want and `getSettings` that returns the rest

Reply:
I disagree with that a bit, since the logic for getting the `*sql.DB` and the instance settings is so intertwined that separate `GetDBFromQuery` and `getSettings` functions would be basically repeated code. I could see the logic in making `GetDBConnectionFromQuery` private again and adding `GetDBFromQuery` that wraps it and just returns the `*sql.DB` if we want to reduce the exposed information.